code/game/objects/items/rcd/RCD.dm

```diff
@@ -349,6 +352,11 @@
 				root_category = new_root
 				update_static_data_for_all_viewers()
 
+		if("select_direction")
+			var/new_dir = text2dir(params["selected_direction"])
+			selected_direction = (isnull(new_dir) || new_dir == selected_direction) ? null : new_dir
```

Comment:
what happens if someone passes us a direction that is not valid? (has more then one cardinal in it/is a real fucking large number)